integration-tests/src/test/java/io/druid/tests/hadoop/ITHadoopIndexTest.java

```diff
@@ -0,0 +1,134 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Guice;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.Callable;
+
+@Guice(moduleFactory = DruidTestModuleFactory.class)
+public class ITHadoopIndexTest extends AbstractIndexerTest
+{
+  private static final Logger LOG = new Logger(ITHadoopIndexTest.class);
+  private static String BATCH_TASK = "/hadoop/batch_hadoop_indexer.json";
+  private static String BATCH_QUERIES_RESOURCE = "/hadoop/batch_hadoop_queries.json";
+  private static String BATCH_DATASOURCE = "batchHadoop";
+  private static boolean dataLoaded = false;
+
+  @Inject
+  private IntegrationTestingConfig config;
+
+  @BeforeClass
+  public void beforeClass()
+  {
+    loadData(config.getProperty ("hadoopTestDir") + "/batchHadoop1");
+    dataLoaded = true;
+  }
+
+  @Test
+  public void testHadoopIndex() throws Exception
+  {
+      queryHelper.testQueriesFromFile(BATCH_QUERIES_RESOURCE, 2);
+  }
+
+  private void loadData(String hadoopDir)
+  {
+    String indexerSpec = "";
+
+    try {
+       LOG.info("indexerFile name: [%s]", BATCH_TASK);
+       indexerSpec = getTaskAsString(BATCH_TASK);
+       indexerSpec = indexerSpec.replaceAll("%%HADOOP_TEST_PATH%%", hadoopDir);
+    } catch (Exception e) {
+      LOG.error ("could not read and modify indexer file: %s", e.getMessage());
+      throw Throwables.propagate(e);
+    }
+
+    try {
+      final String taskID = indexer.submitTask(indexerSpec);
+      LOG.info("TaskID for loading index task %s", taskID);
+      // use longer timeout than waitUntilTaskCompletes() does;
+      // the grid might be slow
+      RetryUtil.retryUntil(
+          new Callable<Boolean>()
+          {
+            @Override
+            public Boolean call() throws Exception
+            {
+              TaskStatus.Status status = indexer.getTaskStatus(taskID);
+              if (status == TaskStatus.Status.FAILED) {
+                throw new ISE("Indexer task FAILED");
+              }
+              return status == TaskStatus.Status.SUCCESS;
+            }
+          },
+          true,
+          60000,
+          20,
+          "Index Task to complete"
+      );
+
+      RetryUtil.retryUntil(
+          new Callable<Boolean>()
+          {
+            @Override
+            public Boolean call() throws Exception
+            {
+              return coordinator.areSegmentsLoaded(BATCH_DATASOURCE);
+
+            }
+          },
+          true,
+          20000,
+          10,
+          "Segment-Load-Task-" + taskID
+      );
+    } catch (Exception e) {
+      LOG.error ("data could not be loaded: %s", e.getMessage());
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @AfterClass
+  public void afterClass()
+  {
+      if (dataLoaded)
+      {
+         try
+         {
+	     unloadAndKillData(BATCH_DATASOURCE);
+         } catch (Exception e) {
+              LOG.warn ("exception while removing segments: " + e.getMessage());
```

Comment:
can you use string pattern instead of concatenation, also pls log the exception too not just the message.